src/main/java/org/cloudburstmc/api/block/behavior/BlockBehavior.java

```diff
@@ -0,0 +1,318 @@
+package org.cloudburstmc.api.block.behavior;
+
+import lombok.val;
+import org.cloudburstmc.api.block.Block;
+import org.cloudburstmc.api.block.BlockState;
+import org.cloudburstmc.api.entity.Entity;
+import org.cloudburstmc.api.item.ItemStack;
+import org.cloudburstmc.api.item.TierType;
+import org.cloudburstmc.api.item.ToolType;
+import org.cloudburstmc.api.player.Player;
+import org.cloudburstmc.api.util.AxisAlignedBB;
+import org.cloudburstmc.api.util.Direction;
+import org.cloudburstmc.api.util.data.BlockColor;
+import org.cloudburstmc.math.vector.Vector3f;
+import org.cloudburstmc.math.vector.Vector3i;
+
+import static org.cloudburstmc.api.block.BlockStates.AIR;
+
+public abstract class BlockBehavior {
+
+    public boolean canHarvestWithHand(BlockState state) {  //used for calculating breaking time
+        val type = state.getType();
+        return type.isDiggable() && type.getToolType() == null && type.getTierType() == null;
+    }
+
+    public boolean isBreakable(BlockState state, ItemStack item) {
+        return state.getType().isDiggable();
+    }
+
+    public int tickRate() {
+        return 10;
+    }
+
+    public int onUpdate(Block block, int type) {
+        return 0;
+    }
+
+    public boolean onActivate(Block block, ItemStack item) {
+        return this.onActivate(block, item, null);
+    }
+
+    public boolean onActivate(Block block, ItemStack item, Player player) {
+        return false;
+    }
+
+    public int getBurnChance(BlockState state) {
+        return state.getType().getBurnChance();
+    }
+
+    public int getBurnAbility(BlockState state) {
+        return state.getType().getBurnAbility();
+    }
+
+    public ToolType getToolType(BlockState state) {
+        return state.getType().getToolType();
+    }
+
+    public TierType getMinimalTier(BlockState state) {
+        return state.getType().getTierType();
+    }
+
+    public boolean checkTool(BlockState state, ItemStack item) {
+        val toolType = getToolType(state);
```

Comment:
oh god...`val`

could we *at least* keep this out of the API? i blame creeperface

Reply:
Yes, it was creeperface that put it in there, I don't see any issue with it.